src/cart/crt_init.c

```diff
@@ -62,52 +62,55 @@ crt_lib_fini(void)
 static void
 dump_envariables(void)
 {
-	int	i;
-	char	*val;
-	char    *envars[] = {"D_PROVIDER",
-			     "D_INTERFACE",
-			     "D_DOMAIN",
-			     "D_PORT",
-			     "CRT_PHY_ADDR_STR",
-			     "D_LOG_STDERR_IN_LOG",
-			     "D_LOG_SIZE",
-			     "D_LOG_FILE",
-			     "D_LOG_FILE_APPEND_PID",
-			     "D_LOG_MASK",
-			     "DD_MASK",
-			     "DD_STDERR",
-			     "DD_SUBSYS",
-			     "CRT_TIMEOUT",
-			     "CRT_ATTACH_INFO_PATH",
-			     "OFI_PORT",
-			     "OFI_INTERFACE",
-			     "OFI_DOMAIN",
-			     "CRT_CREDIT_EP_CTX",
-			     "CRT_CTX_SHARE_ADDR",
-			     "CRT_CTX_NUM",
-			     "D_FI_CONFIG",
-			     "FI_UNIVERSE_SIZE",
-			     "CRT_ENABLE_MEM_PIN",
-			     "FI_OFI_RXM_USE_SRX",
-			     "D_LOG_FLUSH",
-			     "CRT_MRC_ENABLE",
-			     "CRT_SECONDARY_PROVIDER",
-			     "D_PROVIDER_AUTH_KEY",
-			     "D_PORT_AUTO_ADJUST",
-			     "D_POLL_TIMEOUT",
-			     "D_LOG_FILE_APPEND_RANK",
-			     "D_QUOTA_RPCS",
-			     "D_POST_INIT",
-			     "D_POST_INCR",
-			     "DAOS_SIGNAL_REGISTER"};
+	int                i;
+	char              *val;
+	static const char *var_names[] = {"D_PROVIDER",
```

Comment:
Two things here, I can see why these are const but I'm not sure why they'd be static.

However - this list changes reasonably frequently and we want to make that easier, plus at some point I'd like to update the getenv functions to warn if they are used for something not in this list so I think putting this list as a static variable outside of any functions would be better, it would mean we could access it elsewhere without moving it again and it would be easier to add new entries to.

Reply:
- [ ] Move the var_names outside the function

Reply:
> * [x]  Move the var_names outside the function

Fixed with commit a63fb0d2a3032fa22eca37f6052ad69fd3fe2782